### ISSUE TITLE
New box_meta data file is pinning the version of the box

### ISSUE THREAD

**Issue by RohanTalip** [hashicorp/vagrant#10338]
### Vagrant version
Vagrant 2.2.0 and 2.1.5 (and I suspect 2.1.3 and 2.1.4, based on the timeline of the suspected pull request) are affected.

2.1.2 works as expected.

### Host operating system
macOS High Sierra 10.13.6

### Guest operating system
ubuntu/bionic64

### Vagrantfile
Here is a minimal Vagrantfile (I have more configuration that I don't think is relevant to the problem) :

```ruby
required_plugins = %w(vagrant-vbguest)
plugins_to_install = required_plugins.select { |plugin| not Vagrant.has_plugin? plugin }
if not plugins_to_install.empty?
  puts "Installing plugins: #{plugins_to_install.join(' ')}"
  if system "vagrant plugin install #{plugins_to_install.join(' ')}"
    exec "vagrant #{ARGV.join(' ')}"
  else
    abort "Installation of one or more plugins has failed. Aborting."
  end
end

Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/bionic64"
end

```

### Debug output
_Provide a link to a GitHub Gist containing the complete debug output:
https://www.vagrantup.com/docs/other/debugging.html. The debug output should
be very long. Do NOT paste the debug output in the issue, just paste the
link to the Gist._

I didn't see this in time to capture it.  I'll try to reproduce it the next time I see it.

### Expected behavior

On vagrant version 2.2.0 and 2.1.5, running `vagrant box outdated` should have shown that I had version 20181018.0.0 of [ubuntu/bionic64](https://app.vagrantup.com/ubuntu/boxes/bionic64), and that there was a new version available, i.e. 20181024.0.0 .

Also, on the same versions, running `vagrant box update` should have actually updated the box.

### Actual behavior

On vagrant version 2.2.0 and 2.1.5, running `vagrant outdated` resulted in no output to the console, and running `vagrant box update` showed the following:

```
$ vagrant box update
==> default: Checking for updates to 'ubuntu/bionic64'
    default: Latest installed version: 20181018.0.0
    default: Version constraints: 20181008.0.0
    default: Provider: virtualbox
==> default: Box 'ubuntu/bionic64' (v20181018.0.0) is running the latest version.
```

Downgrading to version 2.1.2 fixed the problem:

```
$ vagrant box outdated
Checking if box 'ubuntu/bionic64' is up to date...
A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
available! You currently have version '20181018.0.0'. The latest is version
'20181024.0.0'. Run `vagrant box update` to update.

$ vagrant box update
==> default: Checking for updates to 'ubuntu/bionic64'
    default: Latest installed version: 20181018.0.0
    default: Version constraints: 
    default: Provider: virtualbox
==> default: Updating 'ubuntu/bionic64' with provider 'virtualbox' from version
==> default: '20181018.0.0' to '20181024.0.0'...
==> default: Loading metadata for box 'https://vagrantcloud.com/ubuntu/bionic64'
==> default: Adding box 'ubuntu/bionic64' (v20181024.0.0) for provider: virtualbox
    default: Downloading: https://vagrantcloud.com/ubuntu/boxes/bionic64/versions/20181024.0.0/providers/virtualbox.box
    default: Download redirected to host: cloud-images.ubuntu.com
==> default: Successfully added box 'ubuntu/bionic64' (v20181024.0.0) for 'virtualbox'!
```

I skipped checking versions 2.1.3 and 2.14 due to issue #10161 

The last time I saw this (a few days ago), specifying a specific box to update actually helped, but is a little counterintuitive.  The commands that I ran at that time:

```
$ vagrant box update
==> default: Checking for updates to 'ubuntu/bionic64'
    default: Latest installed version: 20181015.0.0
    default: Version constraints: 20181008.0.0
    default: Provider: virtualbox
==> default: Box 'ubuntu/bionic64' (v20181015.0.0) is running the latest version.

$ vagrant box list
ubuntu/bionic64 (virtualbox, 20181015.0.0)

$ vagrant box update --box ubuntu/bionic64
Checking for updates to 'ubuntu/bionic64'
Latest installed version: 20181015.0.0
Version constraints: > 20181015.0.0
Provider: virtualbox
Updating 'ubuntu/bionic64' with provider 'virtualbox' from version
'20181015.0.0' to '20181018.0.0'...
Loading metadata for box 'https://vagrantcloud.com/ubuntu/bionic64'
Adding box 'ubuntu/bionic64' (v20181018.0.0) for provider: virtualbox
Downloading: https://vagrantcloud.com/ubuntu/boxes/bionic64/versions/20181018.0.0/providers/virtualbox.box
Download redirected to host: cloud-images.ubuntu.com
Successfully added box 'ubuntu/bionic64' (v20181018.0.0) for 'virtualbox'!

$ vagrant box list
ubuntu/bionic64 (virtualbox, 20181015.0.0)
ubuntu/bionic64 (virtualbox, 20181018.0.0)

$ vagrant box outdated --global
* 'ubuntu/bionic64' for 'virtualbox' (v20181018.0.0) is up to date
* 'ubuntu/bionic64' for 'virtualbox' is outdated! Current: 20181015.0.0. Latest: 20181018.0.0

$ vagrant box prune
The following boxes will be kept...
ubuntu/bionic64 (virtualbox, 20181018.0.0)

Checking for older boxes...
Removing box 'ubuntu/bionic64' (v20181015.0.0) with provider 'virtualbox'...
```

I believe the version constraints came from the .vagrant/machines/default/virtualbox/box_meta file.

### Steps to reproduce
1. Install vagrant 2.1.2
2. Create a vagrant installation (e.g. via `vagrant init ubuntu/bionic64`)
3. Run `vagrant up`
4. Run `vagrant halt`
5. Upgrade to vagrant 2.2.0 or 2.1.5 (or probably also 2.1.4 and 2.13)
6. Wait a few days to a week.  Notice that there have been no new prompts to run `vagrant box update` in the output from running `vagrant up` even though there is at least one new version on the [Vagrant Cloud](https://app.vagrantup.com) website.
7. Downgrade to vagrant 2.1.2 and notice that `vagrant box outdated` and `vagrant box update` now work as expected (again).

Installing vagrant 2.1.2 the first time (step 1) is likely optional (but obviously you would have to do `vagrant init` with a later version of vagrant).  It is also likely that there is a way to speed up the steps to reproduce with editing config files and removing certain directories (such as box or machine directories), but I saw no CLI equivalents.

### References
- Issue #10254 seems related.
- Pull request #10083 is suspect, since it added code for the box_meta data file.

**Comment by RohanTalip** [comment]
I should add what the contents of the .vagrant/machines/default/virtualbox/box_meta file are.  In my project where I first saw this problem, the box_meta file contained version 20181008.0.0:
```
{"name":"ubuntu/bionic64","version":"20181008.0.0","provider":"virtualbox","directory":"boxes/ubuntu-VAGRANTSLASH-bionic64/20181008.0.0/virtualbox"}
```

In a new clean directory, using vagrant 2.2.0, I ran the following:
```
$ vagrant init --minimal ubuntu/bionic64
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.
```

No .vagrant/machines/default/virtualbox/box_meta exists yet.

```
$ vagrant box outdated
Checking if box 'ubuntu/bionic64' is up to date...
A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
available! You currently have version '20181024.0.0'. The latest is version
'20181027.0.0'. Run `vagrant box update` to update.
```

No .vagrant/machines/default/virtualbox/box_meta exists yet.

```
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/bionic64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/bionic64' is up to date...
==> default: A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
==> default: available! You currently have version '20181024.0.0'. The latest is version
==> default: '20181027.0.0'. Run `vagrant box update` to update.
...
```

The .vagrant/machines/default/virtualbox/box_meta file for this new machine now contains:
```
{"name":"ubuntu/bionic64","version":"20181024.0.0","provider":"virtualbox","directory":"boxes/ubuntu-VAGRANTSLASH-bionic64/20181024.0.0/virtualbox"}
```

Running `vagrant box outdated` again produces the same result:
```
$ vagrant box outdated
Checking if box 'ubuntu/bionic64' is up to date...
A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
available! You currently have version '20181024.0.0'. The latest is version
'20181027.0.0'. Run `vagrant box update` to update.
```

To recreate the issue in my project, I copied the box_meta file from my project and overwrote the one in this directory, so that it now refers to version 20181008.0.0, and reran `vagrant box outdated`.  There was no output this time.

I created a GitHub Gist for `env VAGRANT_LOG=debug vagrant box outdated`
https://gist.github.com/RohanTalip/bb0dd968bbc45a8cc86cdf0dd092620e

My current boxes:
```
$ vagrant box list
ubuntu/bionic64 (virtualbox, 20181018.0.0)
ubuntu/bionic64 (virtualbox, 20181024.0.0)
```

When I updated the box_meta file to match the oldest version as reported by `vagrant box list`, i.e. 20181018.0.0, then I saw the expected prompt to upgrade again:

```
$ vagrant box outdated
Checking if box 'ubuntu/bionic64' is up to date...
A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
available! You currently have version '20181018.0.0'. The latest is version
'20181027.0.0'. Run `vagrant box update` to update.
```

So, it seems that if the version in the box_meta file doesn't match one that is reported by `vagrant box list` then no update message is posted.  I probably got into this state via upgrading and downgrading vagrant at least once, and running `vagrant box update` and `vagrant box prune` while downgraded.  In the older version of vagrant, the box_meta file would not have been updated since that logic didn't exist yet.

I suggest that the box_meta logic should ignore the box_meta file if it doesn't contain a version from the results of `vagrant box list`, or at least print a warning.

**Comment by airhuman** [comment]
I delete box_meta file, then it works well.
